### PR TITLE
Auto detect common server new

### PR DESCRIPTION
**Pull request by orhovy**
## Status
- [] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Description
Added a new function auto_detect_indicator_type for the issue: demisto/etc#23939
related pr: https://github.com/demisto/content/pull/6969

### CODE REVIEW

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 1]
```diff
@@ -252,6 +258,55 @@ def ip_to_indicator_type(ip):
             return None
 
 
+def auto_detect_indicator_type(indicator_value):
+    """Infer the type of the indicator.
+    Args:
+        indicator_value(str): The indicator whose type we want to check.
+    Returns:
+        str. The type of the indicator.
+    """
```
Comment (bakatzir) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:267]: please use this format:
"""Returns the indicator type of the input IP.
        :type ip: ``str``
        :param ip: IP address to get it's indicator type.
        
        :rtype: ``str``
        :return:: Indicator type from FeedIndicatorType, or None if invalid IP address.

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 2]
```diff
@@ -252,6 +258,58 @@ def ip_to_indicator_type(ip):
...
+      Infer the type of the indicator.
+
+      :type indicator_value: ``str``
+      :param indicator_value: The indicator whose type we want to check. (required)
+
+      :return: The type of the indicator.
+      :rtype: ``str``
+    """
+    if re.match(ipv4cidrRegex, indicator_value):
```
Comment (anara123) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:271]: how about moving the import to here and if we get ImportError then return proper error that explains the user that if he wants to use auto_detect_indicator_type function he should use docker image that contains tldextract module and give an example to the dockerimage we have

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 3]
```diff
@@ -268,6 +263,12 @@ def auto_detect_indicator_type(indicator_value):
       :return: The type of the indicator.
       :rtype: ``str``
     """
+    try:
+        import tldextract
+    except Exception:
+        raise Exception("Missing tldextract module, In order to use the auto detect function please use a docker"
```
Comment (bakatzir) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:269]:
```suggestion
        raise Exception("Missing tldextract module, In order to auto detect indicator types please use a docker"
```

**Comment by teizenman** [comment]
Maybe we should add the `tldextract` package to out basic docker images?

**Comment by orhovy** [comment]
@teizenman  This cant be done as the docker image that common server python uses is a basic image and extern modules should not be installed on it.

**Comment by anara123** [comment]
@orhovy actually we can move it to our demisto:python3 images. It means some of the images will include `tldextract` package and some of the will not.

**Comment by teizenman** [comment]
@anara123 `tldextract` is also supported for Python2, is there a reason not to update the basic demisto:python2 images as well?

**Comment by anara123** [comment]
@teizenman Sure we can update all the 4 images.
demisto/python
demisto/python-deb
demisto/python3
demisto/python3-deb

**Pull request closed without merging**